src/hash.cr

```diff
@@ -353,6 +353,16 @@ class Hash(K, V)
   # as Pointer(UInt16).
   private MAX_INDICES_BYTESIZE_2 = 65536
 
+  # Special value returned by `get_index` to indicate an empty index. Can be
+  # replaced with a used index by `#upsert`. Stops linear entry lookup in
+  # `@indices`.
+  private EMPTY_INDEX = -2
```

Comment:
**suggestion:** *empty* is not an ideal term because deleted indices are also "empty". I was wondering if we could find something better? 
`UNUSED_INDEX` perhaps? It's a subtle difference but refers to the index never being used as opposed to being (currently) unassigned (i.e. empty).

Reply:
The way I see it, "unused" = "empty" + "deleted", rather than "empty" = "unused" + "deleted"